app/models/kubernetes_image_scan.rb

```diff
@@ -0,0 +1,264 @@
+      :wait_pod     => {'creating_pod'       => 'waiting_to_analyze'},
+      :analyze      => {'waiting_to_analyze' => 'analyzing'},
+      :cleanup      => {'analyzing'          => 'removing_pod'},
+      :abort_job    => {'*'                  => 'aborting'},
+      :finish       => {'removing_pod' => 'finished',
+                        'aborting'     => 'finished'},
+    }
+  end
+
+  def initializing
+    queue_signal(:start)
+  end
+
+  def start
+    image = target_entity
+    return queue_signal(:abort_job) unless image
+
+    update!(:options => options.merge(
+      :ems_id          => image.ext_management_system.id,
+      :scan_id         => SecureRandom.uuid,
+      :docker_image_id => image.docker_id,
+      :image_full_name => image.full_name,
+      :pod_name        => "manageiq-img-scan-#{image.docker_id[0..11]}",
+      :pod_port        => INSPECTOR_PORT,
+      :pod_namespace   => INSPECTOR_NAMESPACE
+    ))
+
+    pod = pod_definition
+    _log.info("creating pod #{pod_full_name} to analyze docker image " \
+              "#{options[:docker_image_id]}: #{pod.inspect}")
+
+    begin
+      kubernetes_client.create_pod(pod)
+    rescue KubeException => e
+      _log.info("pod creation for #{pod_full_name} failed: #{e}")
+      return queue_signal(:abort_job)
+    end
+
+    queue_signal(:wait_pod)
+  end
+
+  def wait_pod
+    _log.info("waiting for pod #{pod_full_name} to be available")
+
+    client       = kubernetes_client
+    health_url   = pod_proxy_url(client, INSPECTOR_HEALTH_PATH)
+    http_options = {
+      :use_ssl     => health_url.scheme == 'https',
+      :verify_mode => ext_management_system.verify_ssl_mode
+    }
+
+    loop do
+      response = pod_health_poll(client, health_url, http_options)
+
+      case response
+      when Net::HTTPOK
+        _log.info("pod #{pod_full_name} is ready and accessible")
+        break
+      when Net::HTTPServiceUnavailable
+        # continue: pod is still not up and running
+      else
+        _log.info("unknown access error to pod #{pod_full_name}: " \
+                  "#{response}")
+        return queue_signal(:abort_job)
+      end
+
+      _log.info("pod #{pod_full_name} is not available")
+      sleep(POD_POLL_INTERVAL)
+    end
+
+    queue_signal(:analyze)
+  end
+
+  def analyze
+    image = target_entity
+    return queue_signal(:abort_job) unless image
+
+    _log.info("analyzing image #{options[:docker_image_id]}")
+    # TODO: check that the image id is correct
+
+    client = kubernetes_client
+    content_url = pod_proxy_url(client, INSPECTOR_CONTENT_PATH)
+
+    ost = OpenStruct.new(
+      :uri         => content_url,
+      :verify_mode => ext_management_system.verify_ssl_mode,
+      :headers     => client.headers.stringify_keys,
+      :category    => SCAN_CATEGORIES.join(','),
+      :xml_class   => XmlHash::Document,
+      :guest_os    => KUBERNETES_GUEST_OS
+    )
+
+    image.perform_metadata_scan(ost)
+    image.perform_metadata_sync(ost)
```

Comment:
@roliveri because the code in `scan_metadata` and `.sync_metadata` are very much vm-specific and I am not using an host/smart_proxy for the analysis.